test/config/config.test.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict'
+
+const test = require('tape')
+
+const { CONFIG_DEFINITIONS } = require('./fixtures/config-schema')
+
+const Agent = require('../../lib/agent')
+const { NoopTransport } = require('../../lib/noop-transport')
+const MockLogger = require('./_mock_logger')
+
+// Options to pass to `agent.start()` to turn off some default agent behavior
+// that is unhelpful for these tests.
+const agentOpts = {
+  centralConfig: false,
+  captureExceptions: false,
+  metricsInterval: '0s',
+  cloudProvider: 'none',
+  logLevel: 'warn'
+}
+const agentOptsNoTransportNoLogger = Object.assign(
+  {},
+  agentOpts,
+  {
+    logger: new MockLogger(),
+    transport: function createNoopTransport () {
+      // Avoid accidentally trying to send data to an APM server.
+      return new NoopTransport()
+    }
+  }
+)
+
+function hasType (def, type) {
+  return def.types && def.types.indexOf(type) !== -1
+}
+
+function testEnvVar (t, option, value1, value2) {
+  const existingEnvValue = process.env[option.envVar]
+  const agent = new Agent()
+  const opts = {}
+  let value = value1
+
+  if (value2) {
+    opts[option.name] = value1
+    value = value2
+  }
+
+  process.env[option.envVar] = value.toString()
+  agent.start(agentOptsNoTransportNoLogger)
+  t.strictEqual(agent._conf[option.name], value)
+
+  process.env[option.envVar] = existingEnvValue
+  agent.destroy()
+  t.end()
+}
+
+// Test BOOLS that can be set via ENV
+const BOOL_OPTS_WITH_ENV_EXCLUDED = [
+  // This ones depend on other options so we should have specific tests
+  'captureSpanStackTraces',
+  'asyncHooks',
+  // this one seems to have a bug
+  'breakdownMetrics'
+]
```

Comment:
It would be nice if this was data-driven eventually. I.e. if the set of config vars tested here was from an export of `lib/config/schema.js`. Is that the eventual intention?

I see that is done for `NUM` tests below.

Reply:
I think it might be even better for discussions and improvements to have `CONFIG_SCHEMA` written on a direct export instead of having in a procedural way. I'll keep  the generator code just in case we need to generate it again